src/structure/grid.ts

```diff
@@ -727,6 +710,30 @@ export class ViewersGrid {
 
         return newGUID;
     }
+
+    /**
+     * Function to return *optimal* arrangement of n widgets.
+     * Defaults to a 4 x ? grid. Hardcoded for simplicity.
+     */
+    private bestGridArrangement(n: number) {
+        switch (n) {
+            case 1:
+            case 2:
+                return { rows: n, columns: 1 };
+            case 3:
+            case 4:
+                return { rows: 2, columns: 2 };
+            case 5:
+            case 6:
+                return { rows: 3, columns: 2 };
+            case 8:
+            case 7:
+            case 9:
+                return { rows: 3, columns: 3 };
+            default:
+                throw Error(`Can not create a grid with more than ${this._maxWidgets} elements`);
```

Comment:
This should never be reached, so let's make it clear
```suggestion
                throw Error("reached unreachable code: too many viewer in the grid");
```